lib/chat_list/main_page_tabs/settings_tab.dart

```diff
@@ -68,16 +60,33 @@ class _SettingsTabState extends State<SettingsTab> {
     });
   }
 
+  Widget _settingsWidget(BuildContext context, SettingsModel settingsModel) {
+    if (settingsModel is SimpleSettingsModel) {
+      return _simpleSettings(settingsModel);
+    } else if (settingsModel is BoolSettingsModel) {
+      return _boolSettings(settingsModel);
+    } else if (settingsModel is UserSettingsModel) {
+      return _userSettings(settingsModel);
+    } else if (settingsModel is LogOutSettingsModel) {
+      return _logOutSettings(context, settingsModel);
+    } else if (settingsModel is InfoSettingsModel) {
+      return _infoSettings(context);
+    } else if (settingsModel is LanguageSettingsModel) {
+      return _languageSetting(context, settingsModel);
+    }
+    return null;
+  }
+
   _showLogoutDialog(BuildContext context) {
+    final userRepository = Provider.of<UserRepository>(context, listen: false);
     showDialog(
         context: context,
-        builder: (_) =>
-            AlertDialog(
-              title: Text(LogoutDialogStrings.logoutDialogTitle),
-              content: Text(LogoutDialogStrings.logoutDialogDetails),
+        builder: (_) => AlertDialog(
+              title: Text(S.of(context).logoutDialogTitle),
+              content: Text(S.of(context).logoutDialogDetails),
               actions: <Widget>[
                 ChangeNotifierProvider(
-                  create: (_) => SignOutModel(_userRepository),
```

Comment:
I suppose that overengineering is her. It can be done shortly

Reply:
SignIn and SignOut models united to single AuthModel